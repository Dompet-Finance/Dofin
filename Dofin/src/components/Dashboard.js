'use strict'

import React from 'react'

import {
  View,
  Text,
  TouchableHighlight,
  Button,
} from 'react-native'

class Overview extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Income
        </Text>
        <Text>
          Expenses
        </Text>
        <Text>
          Balance
        </Text>
      </View>
    )
  }
}

class Graph extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Expenses by category
        </Text>
      </View>
    )
  }
}

class Dreams extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Dream come true
        </Text>
      </View>
    )
  }
}

class FapButton extends React.Component {
  render() {
    return (
      <Button
        onPress={() => alert()}
        title="+"
        />
    )
  }
}

export class Dashboard extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View>
        <Overview />
        <Graph />
        <Dreams />
        <FapButton />
      </View>
    )
  }
}

const styles = {
}

export default Dashboard