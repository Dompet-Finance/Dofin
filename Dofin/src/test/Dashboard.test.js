'use strict'

import 'react-native'
import 'react-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Dashboard from '../components/Dashboard.js'

describe('Test Dashboard component', () => {

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Dashboard />)
  })

  test('renders correctly', () => {
    const tree = renderer.create(
      <Dashboard />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('should render overview', () => {
    const child = wrapper.find('Overview')
    expect(child).toHaveLength(1)
  })

  test('should render graph', () => {
    const child = wrapper.find('Graph')
    expect(child).toHaveLength(1)
  })

  test('should render dreams', () => {
    const child = wrapper.find('Dreams')
    expect(child).toHaveLength(1)
  })

  test('should render fap button', () => {
    const child = wrapper.find('FapButton')
    expect(child).toHaveLength(1)
  })

})

