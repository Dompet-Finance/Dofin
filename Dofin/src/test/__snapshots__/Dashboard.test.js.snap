// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard component renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Income
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Expenses
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Balance
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Expenses by category
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Dream come true
    </Text>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
